django/core/mail/message.py

```diff
@@ -280,6 +282,8 @@ def message(self):
             # Use cached DNS_NAME for performance
             msg["Message-ID"] = make_msgid(domain=DNS_NAME)
         for name, value in self.extra_headers.items():
+            if name.lower() in HEADER_WITH_ONE_OCCURRENCE and msg[name] is not None:
```

Comment:
Why the `msg[name] is not None` condition? :)

Reply:
Yeah, I would outright forbid the headers that can be set via our public API and not only if they are not set otherwise. It might also be worth to move it up to `__init__` then?

Reply:
Mhm looking at https://code.djangoproject.com/ticket/17444 we have to be very careful with this change. Looks like we explicitly want to support overriding `to` via `extra_headers` :/

Reply:
> Why the `msg[name] is not None` condition? :)

I'm sorry for that. It is a mistake